tests/test_taskprocessor.c

```diff
@@ -963,6 +966,174 @@ AST_TEST_DEFINE(serializer_pool)
+	tps = ast_taskprocessor_get("test_cli_taskprocessor", TPS_REF_DEFAULT);
+	if (!tps) {
+		ast_test_status_update(test, "Unable to create test taskprocessor\n");
+		close(cli_output_fd[0]);
+		close(cli_output_fd[1]);
+		return AST_TEST_FAIL;
+	}
+
+	/* Create tasks that will wait so they stay in the queue */
+	task_data1 = task_data_create();
+	task_data2 = task_data_create();
+	task_data3 = task_data_create();
+
+	if (!task_data1 || !task_data2 || !task_data3) {
+		ast_test_status_update(test, "Unable to create task_data\n");
+		goto cleanup;
+	}
+
+	/* Set a long wait time so tasks stay queued */
+	task_data1->wait_time = 5000; /* 5 seconds */
+	task_data2->wait_time = 5000;
+	task_data3->wait_time = 5000;
+
+	/* Queue the tasks */
+	if (ast_taskprocessor_push(tps, task, task_data1)) {
+		ast_test_status_update(test, "Failed to queue task 1\n");
+		goto cleanup;
+	}
+	if (ast_taskprocessor_push(tps, task, task_data2)) {
+		ast_test_status_update(test, "Failed to queue task 2\n");
+		goto cleanup;
+	}
+	if (ast_taskprocessor_push(tps, task, task_data3)) {
+		ast_test_status_update(test, "Failed to queue task 3\n");
+		goto cleanup;
+	}
+
+	/* Give the taskprocessor a moment to start processing */
+	usleep(100000); /* 100ms */
+
+	/* Execute the CLI command */
+	snprintf(cli_command, sizeof(cli_command), "core show taskprocessor name test_cli_taskprocessor");
+
+	if (ast_cli_command(cli_output_fd[1], cli_command) != 0) {
+		ast_test_status_update(test, "CLI command execution failed\n");
+		goto cleanup;
+	}
+
+	/* Close write end and read the output */
+	close(cli_output_fd[1]);
+	cli_output_fd[1] = -1;
+
+	bytes_read = read(cli_output_fd[0], output_buffer, sizeof(output_buffer) - 1);
+	if (bytes_read <= 0) {
+		ast_test_status_update(test, "Failed to read CLI output\n");
+		goto cleanup;
+	}
+	output_buffer[bytes_read] = '\0';
+
+	/* Log the output for inspection */
+	ast_test_status_update(test, "CLI Output:\n%s\n", output_buffer);
+
+	/* Verify the output contains expected information */
+	if (!strstr(output_buffer, "test_cli_taskprocessor")) {
+		ast_test_status_update(test, "Output missing taskprocessor name\n");
+		goto cleanup;
+	}
+
+	if (!strstr(output_buffer, "Current queue size")) {
+		ast_test_status_update(test, "Output missing queue size information\n");
+		goto cleanup;
+	}
+
+	/* Check for queued tasks section (at least one task should be shown) */
+	if (!strstr(output_buffer, "Queued Tasks") && !strstr(output_buffer, "Currently executing")) {
+		ast_test_status_update(test, "Output missing queued tasks or execution status\n");
+		goto cleanup;
+	}
+
+	/* Verify we see task information */
+	if (!strstr(output_buffer, "Task #")) {
+		ast_test_status_update(test, "Output missing task information\n");
+		goto cleanup;
+	}
+
+	ast_test_status_update(test, "CLI command output validated successfully\n");
+	res = AST_TEST_PASS;
+
+cleanup:
+	/* Signal all tasks to complete */
+	if (task_data1) {
+		ast_mutex_lock(&task_data1->lock);
```

Comment:
Why does this logic exist to set wait_time to 0? The tasks aren't in a loop, they wait a single time and altering wait_time isn't going to make them wake up early.